File: rest-service/src/main/java/mb/lib/formatter/db/InsertReportRow.java
package mb.lib.formatter.db;

import java.sql.Connection;
import java.sql.PreparedStatement;

import io.vulpine.lib.query.util.basic.BasicPreparedVoidQuery;
import mb.lib.formatter.model.ReportRow;
import mb.lib.util.JSON;

public class InsertReportRow
{
  private static final String Query = """
    INSERT INTO
      userlogins5.multiblast_fmt_jobs (
        report_digest
      , job_digest
      , status
      , config
      , queue_id
      )
    VALUES
      (?, ?, ?, ?, ?)
    """;

  private final Connection con;
  private final ReportRow row;

  public InsertReportRow(Connection con, ReportRow row) {
    this.con = con;
    this.row = row;
  }

  public void run() throws Exception {
    new BasicPreparedVoidQuery(Query, con, this::prepare).execute();
  }

  private void prepare(PreparedStatement ps) throws Exception {
    ps.setBytes(1, row.getReportID().bytes());
    ps.setBytes(2, row.getJobID().bytes());
    ps.setString(3, row.getStatus().getValue());
    ps.setString(4, JSON.stringify(row.getConfig()));
    ps.setInt(5, row.getQueueID());
  }
}

File: rest-service/src/main/kotlin/mb/lib/report/db/InsertReportRow.kt
package mb.lib.report.db

import io.vulpine.lib.query.util.basic.BasicPreparedVoidQuery
import mb.lib.report.model.ReportRow
import mb.lib.util.jsonStringify
import java.sql.Connection
import java.sql.PreparedStatement

data class InsertReportRow(
  val con: Connection,
  val row: ReportRow
) {
  companion object {
    @Suppress("SpellCheckingInspection")
    // language=postgresql
    private const val Query = """
    INSERT INTO
      multiblast.multiblast_fmt_jobs (
        report_digest
      , job_digest
      , status
      , config
      , queue_id
      )
    VALUES
      (?, ?, ?, ?, ?)
    ON CONFLICT DO NOTHING
    """
  }

  fun run() = BasicPreparedVoidQuery(Query, con, this::prepare).execute()

  private fun prepare(ps: PreparedStatement) {
    ps.setBytes(1, row.reportID.bytes)
    ps.setBytes(2, row.jobID.bytes)
    ps.setString(3, row.status.value)
    ps.setString(4, row.config.jsonStringify())
    ps.setInt(5, row.queueID)
  }
}
